server/metrics/metrics.go

```diff
@@ -0,0 +1,109 @@
+	CacheTypeLabel = "cache_type"
+
+	// TODO(bduffany): Explain the difference between `miss` and `upload`
+	/// Cache event type: `hit`, `miss`, or `upload`.
+	CacheEventTypeLabel = "cache_event_type"
+)
+
+var (
+	/// ## Invocation metrics (builds and tests)
+	///
+	/// NOTE: Invocation metrics are recorded at the end of each invocation,
+	/// which means that these metrics provide _approximate_ real-time signals.
+
+	InvocationDurationUs = promauto.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: "buildbuddy",
+		Subsystem: "invocation",
+		Name:      "duration_us",
+		Buckets:   prometheus.ExponentialBuckets(1, 10, 9),
+		Help:      "The total duration of each invocation, in **microseconds**.",
+	}, []string{
+		// TODO: Slice on build vs. test
+		InvocationStatusLabel,
+	})
+
+	/// ## Remote cache metrics
+	///
+	/// NOTE: Cache metrics are recorded at the end of each invocation,
+	/// which means that these metrics provide _approximate_ real-time signals.
+
+	CacheEvents = promauto.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: "buildbuddy",
+		Subsystem: "remote_cache",
+		Name:      "events",
+		Buckets:   prometheus.ExponentialBuckets(1, 10, 9),
+		Help:      "Number of cache events handled in each invocation. Use the **`_sum`** suffix to get the total number of cache events across all invocations.",
+	}, []string{
+		InvocationStatusLabel,
+		CacheTypeLabel,
+		CacheEventTypeLabel,
+	})
+
+	CacheDownloadSizeBytes = promauto.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: "buildbuddy",
+		Subsystem: "remote_cache",
+		Name:      "download_size_bytes",
+		Buckets:   prometheus.ExponentialBuckets(1, 10, 9),
+		Help:      "Number of bytes downloaded from the remote cache per invocation. Use the **`_sum`** suffix to get the total downloaded bytes across all invocations.",
+	}, []string{
+		InvocationStatusLabel,
+	})
+
+	CacheUploadSizeBytes = promauto.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: "buildbuddy",
+		Subsystem: "remote_cache",
+		Name:      "upload_size_bytes",
+		Buckets:   prometheus.ExponentialBuckets(1, 10, 9),
+		Help:      "Number of bytes uploaded to the remote cache per invocation. Use the **`_sum`** suffix to get the total uploaded bytes across all invocations.",
+	}, []string{
+		InvocationStatusLabel,
+	})
+
+	/// ## Internal metrics
+	///
+	/// These metrics are for monitoring lower-level subsystems of BuildBuddy.
+	///
+	/// ### Build event handler
+	///
+	/// The build event handler logs all build events uploaded to BuildBuddy
+	/// as part of the Build Event Protocol.
+
+	BuildEventHandlerDurationUs = promauto.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: "buildbuddy",
+		Subsystem: "build_event_handler",
+		Name:      "duration_us",
+		Buckets:   prometheus.ExponentialBuckets(1, 10, 9),
+		Help:      "The time spent handling each build event in **microseconds**. Use the **`_count`** suffix to get the total number of build events handled.",
```

Comment:
nit: if this is an internal metric can we remove the "Use the **`_count`** suffix to get the total number of build events handled." ?

Reply:
Done